test/Reducer.modifyStack.test.js

```diff
@@ -0,0 +1,378 @@
+      });
+
+      const action = {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          {type: ActionConst.ModifyStackTypes.REMOVE, sceneKey: 'termsOfService'},
+        ],
+      };
+      expect(() => reducer(latestState, action)).to.throw(/could not be found in the stack/);
+    });
+
+    it('cannot no sceneKey or index', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      const action = {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          {type: ActionConst.ModifyStackTypes.REMOVE},
+        ],
+      };
+      expect(() => reducer(latestState, action)).to.throw(/sceneKey or index has to be defined/);
+    });
+  });
+
+  describe('INSERT', () => {
+    it('to root stack, no position', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService' },
+        ],
+      });
+
+      expect(latestState.index).equal(2);
+      expect(latestState.children[0].sceneKey).equal('termsOfService');
+      expect(latestState.children[1].sceneKey).equal('sideMenu');
+      expect(latestState.children[2].sceneKey).equal('privacyPolicy');
+
+      currentScene = getCurrent(latestState);
+      expect(currentScene.sceneKey).equal('privacyPolicy');
+      expect(currentScene.type).equal(ActionConst.PUSH);
+    });
+
+    it('with index to root stack', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', index: 1 },
+        ],
+      });
+
+      expect(latestState.index).equal(2);
+      expect(latestState.children[0].sceneKey).equal('sideMenu');
+      expect(latestState.children[1].sceneKey).equal('termsOfService');
+      expect(latestState.children[2].sceneKey).equal('privacyPolicy');
+
+      currentScene = getCurrent(latestState);
+      expect(currentScene.sceneKey).equal('privacyPolicy');
+      expect(currentScene.type).equal(ActionConst.PUSH);
+    });
+
+    it('with beforeSceneKey to root stack', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', beforeSceneKey: 'privacyPolicy' },
```

Comment:
`line 242, col 1, Error - Line 242 exceeds the maximum line length of 100. (max-len)`